mono/metadata/w32socket-unix.c

```diff
@@ -1472,11 +1473,12 @@ mono_w32socket_convert_error (gint error)
 #endif
 	case EINTR: return WSAEINTR;
 	case EINVAL: return WSAEINVAL;
-	/*FIXME: case EIO: return WSAE????; */
+	case EIO: return WSA_INVALID_HANDLE; /* not a precise match, best we can do. */
```

Comment:
Why this over WSAEBADF?

Reply:
https://github.com/mono/mono/issues/16024#issuecomment-522653506 it's what w32error-unix.c does, kind of.